### DISCUSSION TITLE
apple music api

### DISCUSSION THREAD

**Discussion by ByAhri** [TopiSenpai/LavaSrc#83] [Q&A] [answered]
I was testing getting access to apple music api with a simple js script but I am always getting an unauthorized response 401
Honestly I am not sure how to do this properly for apple because I haven't been able to get access I tried these things and
I want to make sure if it's something I am doing wrong or if my token is invalid. this token was supposed to be valid for 12 hours

I ask here cause I know that you use apple api too if you could help me I would appreciate it

eyJhbGciOiJFUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE2ODg3ODc1NDYsImlhdCI6MTY4ODc0NDM0Nn0.ndEvBoOOZ1oo--uEfvBGfFm99OF43I7Ko3rfzdeQDKmwDIRDbRkIbjNUakjTurNBXXnVWo9jaIatmW41lx4elw

```javascript
let authOptions = {
    url: "https://api.music.apple.com/v1/catalog/US/search?term=positions&types=songs",
    headers: {
        Authorization: "Bearer eyJhbGciOiJFUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE2ODg3ODc1NDYsImlhdCI6MTY4ODc0NDM0Nn0.ndEvBoOOZ1oo--uEfvBGfFm99OF43I7Ko3rfzdeQDKmwDIRDbRkIbjNUakjTurNBXXnVWo9jaIatmW41lx4elw",
    },
    form: {
        grant_type: "client_credentials",
    },
    json: true,
};
request.post(authOptions, async function (error, response, body) {
    if (!error && response.statusCode === 200) {
        console.log(body)
    }
    else console.log(response.statusCode)
});
```
```javascript
const res = await axios.get(`https://api.music.apple.com/v1/catalog/US/search?term=positions&types=songs`, {
    headers: {
        Authorization: "Bearer eyJhbGciOiJFUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE2ODg3ODc1NDYsImlhdCI6MTY4ODc0NDM0Nn0.ndEvBoOOZ1oo--uEfvBGfFm99OF43I7Ko3rfzdeQDKmwDIRDbRkIbjNUakjTurNBXXnVWo9jaIatmW41lx4elw"
    }
})
console.log(res);
```
I also tried other endpoints to see if I messed up something with the search one but the same result
`https://api.music.apple.com/v1/catalog/US/songs/1553944254`

**Comment by topi314** [comment, marked as answer]
apple music is a pain, idk, have to figure out yourself
